### PR TITLE
Create EnvironmentVarlist from dictionary

### LINKED ISSUE #4444: Create EnvironmentVarlist from dictionary
**Issue**

Move the method from res_config.py:

```python
    @staticmethod
    def _read_env_vars(config_dict):
        env_vars = EnvironmentVarlist()

        environment_vars = config_dict.get(ConfigKeys.SETENV, [])

        for item in iter(environment_vars):
            if isinstance(item, Mapping):
                key = item["NAME"]
                value = item["VALUE"]
            else:
                key, value = item
            env_vars.setenv(key, value)

        paths = config_dict.get("UPDATE_PATH", [])

        for key, value in iter(paths):
            env_vars.update_path(key, value)

        return env_vars
```

inside a classmethod in `EnvironmentVarlist` something like:

```python 
   @classmethod
    def from_dict(config_dict):
        env_vars = EnvironmentVarlist()

        for key, val in iter(config_dict.get(ConfigKeys.SETENV, []):
            env_vars.setenv(key, value)

        for key, value in iter(config_dict.get("UPDATE_PATH", [])):
            env_vars.update_path(key, value)

        return env_vars
```

### PR DESCRIPTION
**Pull request by DanSava**
**Issue**
Resolves #4444 

## Pre review checklist

- [ ] Added appropriate release note label
- [x] PR title captures the intent of the changes, and is fitting for release notes.
- [x] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).

Adding labels helps the maintainers when writing release notes. This is the [list of release note labels](https://github.com/equinor/ert/labels?q=release-notes).

### CODE REVIEW

**Review by valentin-krasontovitsch — COMMENTED** [review]
looks good already, gonna look through those additional tests as well once you add them

File: src/ert/_c_wrappers/job_queue/environment_varlist.py [hunk 1]
```diff
@@ -18,21 +18,27 @@ class EnvironmentVarlist(BaseCClass):
 
     def __init__(
         self,
-        env_vars: Optional[Dict[str, str]] = None,
-        paths: Optional[Dict[str, str]] = None,
+        env_vars: Optional[List[Any]] = None,
+        paths: Optional[List[Any]] = None,
```
[the remark this answers is not in the document]
Comment (DanSava) [reply src/ert/_c_wrappers/job_queue/environment_varlist.py:22]: It is either a tuple or a list of two elements, so I went with `Any`

[the remark this answers is not in the document]
Comment (valentin-krasontovitsch) [reply src/ert/_c_wrappers/job_queue/environment_varlist.py:22]: oh we still have several possibilities, that's too bad. thanks for clarifying!

File: tests/unit_tests/c_wrappers/test_integration_config.py [hunk 2]
```diff
@@ -44,3 +48,43 @@ def test_num_cpu_subst(monkeypatch, tmp_path, append, numcpu):
...
+        path_env_var = os.getenv("PATH")
+        try:
+            config = ResConfig("test.ert")
+            enkf_main = EnKFMain(config)
+            _create_runpath(enkf_main)
+        except Exception as e:
+            if "MYVAR" in os.environ:
+                del os.environ["MYVAR"]
+            raise e
+        del os.environ["MYVAR"]
```
Comment (valentin-krasontovitsch) [inline tests/unit_tests/c_wrappers/test_integration_config.py:75]: could we save a line by putting the deletion of the env var in a finally block?

**Review by valentin-krasontovitsch — APPROVED** [review]
looking good : )

**Pull request merged**